### PR TITLE
Review

### PR DESCRIPTION
**Pull request by anjalidk**
QA infra for ROUTE.

### CODE REVIEW

File: tests/MEM/mem.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# Enable -DTESTING in Makefile and the recompile library - make
```
Comment (hallyn) [inline tests/MEM/mem.sh]: The relative paths here make me nervous.  Can you use the
git rev-parse --show-toplevel to find the top of the git tree
and use paths relative to that?

File: net_route.c [hunk 2]
```diff
@@ -69,6 +69,73 @@ static int send_route_req(int net_sock)
 	return 0;
 }
 
+#ifdef TESTING
+#define IPA_F "%hhu.%hhu.%hhu.%hhu"
+#define IPA_V(str) str[0], str[1], str[2], str[3]
+
+static inline int is_zero(char *str, int len)
```
Comment (hallyn) [inline net_route.c:76]: Would be great to make this return bool.

File: tests/CPU/cpu_test.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+int main(int argc, char **argv)
+{
+        struct cpuinfo *cpu;
+	int cpu_num, i;
+	FILE *fp;
+	int ret;
+
+	cpu_num = sysconf(_SC_NPROCESSORS_ONLN);
+	cpu = (struct cpuinfo *) malloc(cpu_num*sizeof(struct cpuinfo));
```
Comment (hallyn) [inline tests/CPU/cpu_test.c:33]: Even in test cases, it's best to check return values.

**Review by hallyn — COMMENTED** [review]
Just a few small comments.  The shell one applies to all the test .sh files.

**Comment by anjalidk** [comment]
> On Aug 1, 2023, at 6:53 AM, Serge Hallyn ***@***.***> wrote:
> 
> 
> @hallyn commented on this pull request.
> 
> In tests/MEM/mem.sh:
> 
> > @@ -0,0 +1,12 @@
> +#!/bin/sh
> +
> +# Enable -DTESTING in Makefile and the recompile library - make
> 
> The relative paths here make me nervous. Can you use the
> git rev-parse --show-toplevel to find the top of the git tree
> and use paths relative to that?
> 
Thanks, done for all .sh files and did a push.

**Comment by anjalidk** [comment]
> On Aug 1, 2023, at 6:54 AM, Serge Hallyn ***@***.***> wrote:
> 
> 
> @hallyn commented on this pull request.
> 
> In net_route.c:
> 
> > @@ -69,6 +69,73 @@ static int send_route_req(int net_sock)
>  	return 0;
>  }
>  
> +#ifdef TESTING
> +#define IPA_F "%hhu.%hhu.%hhu.%hhu"
> +#define IPA_V(str) str[0], str[1], str[2], str[3]
> +
> +static inline int is_zero(char *str, int len)
> 
> Would be great to make this return bool.
Done and pushed.

**Comment by anjalidk** [comment]
> On Aug 1, 2023, at 6:55 AM, Serge Hallyn ***@***.***> wrote:
> 
> 
> @hallyn commented on this pull request.
> 
> In tests/CPU/cpu_test.c:
> 
> > +
> +#include <stdio.h>
> +#include <stdlib.h>
> +#include <string.h>
> +#include <unistd.h>
> +#include <resource.h>
> +
> +int main(int argc, char **argv)
> +{
> +        struct cpuinfo *cpu;
> +	int cpu_num, i;
> +	FILE *fp;
> +	int ret;
> +
> +	cpu_num = sysconf(_SC_NPROCESSORS_ONLN);
> +	cpu = (struct cpuinfo *) malloc(cpu_num*sizeof(struct cpuinfo));
> 
> Even in test cases, it's best to check return values.
Done & pushed.

**Comment by anjalidk** [comment]
> On Aug 1, 2023, at 6:57 AM, Serge Hallyn ***@***.***> wrote:
> 
> 
> @hallyn commented on this pull request.
> 
> Just a few small comments. The shell one applies to all the test .sh files.
> 
Yes, took care of all comments.

**Comment by hallyn** [comment]
This branch was a bit messy.  I cherrypicked the commits and pushed the result, so this PR is effectively merged.

**Comment by anjalidk** [comment]
Thanks a lot!

**Pull request closed without merging**